src/execution_plan/optimizations/utilize_indices.c

```diff
@@ -181,241 +97,75 @@ bool _simple_predicates(FT_FilterNode *filter) {
-					RediSearch_QueryNodeAddChild(node, right);
-					break;
-				default:
-					ASSERT(false && "unexpected conditional operation");
-					break;
-			}
+		if(AR_EXP_IsAttribute(lhs_exp, NULL)) exp = rhs_exp; // n.v = exp
+		if(AR_EXP_IsAttribute(rhs_exp, NULL)) exp = lhs_exp; // exp = n.v
+		// filter is not of the form n.v = exp or exp = n.v
+		if(exp == NULL) {
+			res = false;
 			break;
 		}
-		case FT_N_PRED: {
-			double d;
-			char *field;
-			bool attribute = AR_EXP_IsAttribute(filter->pred.lhs, &field);
-			ASSERT(attribute == true);
-
-			SIValue v = filter->pred.rhs->operand.constant;
-			switch(SI_TYPE(v)) {
-				case T_STRING:
-					parent = RediSearch_CreateTagNode(sp, field);
-					switch(filter->pred.op) {
-						case OP_LT:    // <
-							node = RediSearch_CreateLexRangeNode(sp, field, RSLEXRANGE_NEG_INF, v.stringval, 0, 0);
-							break;
-						case OP_LE:    // <=
-							node = RediSearch_CreateLexRangeNode(sp, field, RSLEXRANGE_NEG_INF, v.stringval, 0, 1);
-							break;
-						case OP_GT:    // >
-							node = RediSearch_CreateLexRangeNode(sp, field, v.stringval, RSLECRANGE_INF, 0, 0);
-							break;
-						case OP_GE:    // >=
-							node = RediSearch_CreateLexRangeNode(sp, field, v.stringval, RSLECRANGE_INF, 1, 0);
-							break;
-						case OP_EQUAL:  // ==
-							node = RediSearch_CreateTokenNode(sp, field, v.stringval);
-							break;
-						case OP_NEQUAL: // !=
-							ASSERT(false && "Index can't utilize the 'not equals' operation.");
-							break;
-						default:
-							ASSERT(false && "unexpected operation");
-							break;
-					}
-
-					RediSearch_QueryNodeAddChild(parent, node);
-					node = parent;
-					break;
-
-				case T_DOUBLE:
-				case T_INT64:
-				case T_BOOL:
-					d = SI_GET_NUMERIC(v);
-					switch(filter->pred.op) {
-						case OP_LT:    // <
-							node = RediSearch_CreateNumericNode(sp, field, d, RSRANGE_NEG_INF, false, false);
-							break;
-						case OP_LE:    // <=
-							node = RediSearch_CreateNumericNode(sp, field, d, RSRANGE_NEG_INF, true, false);
-							break;
-						case OP_GT:    // >
-							node = RediSearch_CreateNumericNode(sp, field, RSRANGE_INF, d, false, false);
-							break;
-						case OP_GE:    // >=
-							node = RediSearch_CreateNumericNode(sp, field, RSRANGE_INF, d, false, true);
-							break;
-						case OP_EQUAL:  // ==
-							node = RediSearch_CreateNumericNode(sp, field, d, d, true, true);
-							break;
-						case OP_NEQUAL: // !=
-							ASSERT(false && "Index can't utilize the 'not equals' operation.");
-							break;
-						default:
-							ASSERT(false && "unexpected operation");
-							break;
-					}
-					break;
-				default:
-					ASSERT(false && "unexpected value type");
-					break;
-			}
 
-			break;
-		}
-		case FT_N_EXP: {
-			node = _filterTreeToInQueryNode(filter, sp);
-			break;
-		}
-		default: {
-			ASSERT("unknown filter tree node type");
-			break;
+		// make sure 'exp' represents a scalar
```

Comment:
Maybe `determine whether` instead of `make sure`.

Reply:
👍